content/en/security/application_security/enabling/single_step/threat_detection.md

```diff
@@ -0,0 +1,420 @@
+{{< highlight yaml "hl_lines=7-12" >}}
+   datadog:
+     apiKeyExistingSecret: datadog-secret
+     site: <DATADOG_SITE>
+     apm:
+       instrumentation:
+         enabled: true
+         libVersions: # Add any versions you want to set
+            dotnet: v2.46.0
+            python: v1.20.6
+            java: v1.22.0
+            js: v4.17.0
+            ruby: v1.15.0
+{{< /highlight >}}
+
+Supported languages include:
+
+- .NET (`dotnet`)
+- Python (`python`)
+- Java (`java`)
+- Node.js (`js`)
+- Ruby (`ruby`)
+
+### Tagging observability data by environment {#env-k8}
+
+Automatically tag instrumented services and other telemetry that pass through the Agent with a specific environment. For example, if the Agent is installed in your staging environment, set `env:staging` to associate your observability data with `staging`.
+
+For example, add the following configuration to your `datadog-values.yaml` file:
+{{< highlight yaml "hl_lines=4-5" >}}
+   datadog:
+     apiKeyExistingSecret: datadog-secret
+     site: <DATADOG_SITE>
+     tags:
+         - env:staging
+     apm:
+       instrumentation:
+         enabled: true
+     env:
+      - name: DD_APPSEC_ENABLED
+        value: "true"
+{{< /highlight >}}
+
+[15]: https://github.com/DataDog/dd-trace-dotnet/releases
+{{% /tab %}}
+{{< /tabs >}}
+
+## Removing Single Step APM and ASM instrumentation from your Agent
+
+If you don't want to collect trace data for a particular service, host, VM, or container, complete the follow steps:
+
+### Removing instrumentation for specific services
+
+Run the following commands and restart the service to stop injecting the library into the service and stop producing traces from that service.
+
+{{< tabs >}}
+{{% tab "Linux host or VM" %}}
+
+1. Add the `DD_INSTRUMENT_SERVICE_WITH_APM` environment variable to the service startup command:
+
+   ```shell
+   DD_INSTRUMENT_SERVICE_WITH_APM=false <service_start_command>
+   ```
+2. Restart the service.
+
+3. To disable ASM, remove the `DD_APPSEC_ENABLED=true` environment variable from your application configuration, and restart your service.
+
+
+
+{{% /tab %}}
+
+{{% tab "Docker" %}}
+
+1. Add the `DD_INSTRUMENT_SERVICE_WITH_APM` environment variable to the service startup command:
+   ```shell
+   docker run -e DD_INSTRUMENT_SERVICE_WITH_APM=false <service_start_command>
+   ```
+2. Restart the service.
+
+3. To disable ASM, remove the `DD_APPSEC_ENABLED=true` environment variable from your application configuration, and restart your service.
+{{% /tab %}}
+
+{{% tab "Kubernetes" %}}
+
+1. Set the `admission.datadoghq.com/enabled:` label to `"false"` for the pod spec:
+   ```yaml
+   spec:
+     template:
+       metadata:
+         labels:
+           admission.datadoghq.com/enabled: "false"
+   ```
+2. Apply the configuration:
+   ```bash
+   kubectl apply -f /path/to/your/deployment.yaml
+   ```
+3. Restart the services you want to remove instrumentation for.
+
```

Comment:
I believe it is worth mentioning that they can keep APM working and only disable ASM by adding `DD_APPSEC_ENABLED=false` in their deployments. Something like this:
```suggestion
3. Restart the services you want to remove instrumentation for.

Note that you can disable ASM while keeping APM up by adding the `DD_APPSEC_ENABLED=false` environment variable to your deployments.

```

Ofc feel free to rework this